src/frontend/src/components/public/PIAFormTabs/PIACollectionUseAndDisclosure/index.tsx

```diff
@@ -0,0 +1,208 @@
+import { useState } from 'react';
+import Messages from './messages';
+import MDEditor from '@uiw/react-md-editor';
+import InputText from '../../../common/InputText/InputText';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faPlus, faTrash } from '@fortawesome/free-solid-svg-icons';
+import { useNavigate, useOutletContext, useParams } from 'react-router-dom';
+import { isMPORole } from '../../../../utils/helper.util';
+import { IPiaForm } from '../../../../types/interfaces/pia-form.interface';
+import { PiaStateChangeHandlerType } from '../../../../pages/PIAIntakeForm';
+
+import {
+  CollectionNoticeInput,
+  ICollectionUseAndDisclosure,
+  StepInput,
+} from './CollectionUseAndDisclosure';
+import List, { InputTextProps } from '../../../common/List';
+
+const PIACollectionUseAndDisclosure = () => {
+  const [pia, piaStateChangeHandler] =
+    useOutletContext<[IPiaForm, PiaStateChangeHandlerType]>();
+
+  const [collectionUseAndDisclosureForm, setCollectionUseAndDisclosureForm] =
+    useState<ICollectionUseAndDisclosure>(
+      pia.collectionUseAndDisclosure || {
+        steps: [],
+        collectionNotice: {},
+      },
+    );
+
+  const stateChangeHandler = (
+    value: any,
+    key: keyof ICollectionUseAndDisclosure,
+  ) => {
+    setCollectionUseAndDisclosureForm((state) => ({
+      ...state,
+      [key]: value,
+    }));
+    piaStateChangeHandler(
+      collectionUseAndDisclosureForm,
+      'collectionUseAndDisclosure',
+    );
+  };
+  const [disclosure, setDisclosure] = useState('');
+  const [steps, setSteps] = useState<Array<StepInput>>(
+    collectionUseAndDisclosureForm?.steps.length > 0
+      ? collectionUseAndDisclosureForm?.steps
+      : [
+          {
+            drafterInput: '',
+            mpoInput: '',
+            foippaInput: '',
+            OtherInput: '',
+          },
+          {
+            drafterInput: '',
+            mpoInput: '',
+            foippaInput: '',
+            OtherInput: '',
+          },
+          {
+            drafterInput: '',
+            mpoInput: '',
+            foippaInput: '',
+            OtherInput: '',
+          },
+          {
+            drafterInput: '',
+            mpoInput: '',
+            foippaInput: '',
+            OtherInput: '',
+          },
+        ],
+  );
+  const [MPOCommentsDisclosure, setMPOCommentsDisclosure] = useState('');
+  const [collectionNotice, setCollectionNotice] =
+    useState<CollectionNoticeInput>({
+      drafterInput: '',
+      mpoInput: '',
+    });
+  const [rows, setRows] = useState<Array<InputTextProps[]>>(
+    steps.map((step, i) => [
+      { label: `Step ${i}`, value: step.drafterInput, id: 'one' },
+      { value: step.mpoInput, id: 'two' },
+      { value: step.foippaInput, id: 'three' },
+      { value: step.OtherInput, id: 'four' },
+    ]),
+  );
+
+  const addRow = () => {
```

Comment:
I feel like this method, along with the 'removeRow' method, should be encapsulated within the List component itself. Please correct me if I am wrong, but this way of doing things will require us to redefine 'addRow' and 'removeRow' for every tab that uses the List.

Reply:
Yes, you are right, already create a (sub-task)[https://apps.itsm.gov.bc.ca/jira/browse/UTOPIA-854]